### PR TITLE
Dega api integration

### PR DESCRIPTION
**Pull request by shreeharsha-factly**

### CODE REVIEW

**Review by hetjagani — APPROVED** [review]

File: server/action/video/create.go [hunk 1]
```diff
@@ -372,3 +387,26 @@ func getPublishPermissions(oID, sID, uID int) (int, error) {
 
 	return resStatus, nil
 }
+
+func getRatingAndClaimantIDs(claims []model.Claim) ([]uint, []uint) {
+
+	ratingIDs := make([]uint, 0)
+	claimantIDs := make([]uint, 0)
+
+	rMap := make(map[uint]uint)
```
Comment (hetjagani) [inline server/action/video/create.go:396]: What is purpose of this map. We can directly add in ratingIDs & claimantIDs

File: server/action/video/update.go [hunk 2]
```diff
@@ -286,6 +278,70 @@ func update(w http.ResponseWriter, r *http.Request) {
...
+
+		for _, rating := range ratings {
+			ratingMap[rating.ID] = rating
+		}
+		for _, claimant := range claimants {
+			claimantMap[claimant.ID] = claimant
+		}
+	}
+
+	log.Println(ratingIds)
```
Comment (hetjagani) [inline server/action/video/update.go]: Remove prints

File: server/action/video/update.go [hunk 3]
```diff
@@ -286,6 +278,70 @@ func update(w http.ResponseWriter, r *http.Request) {
...
 	}
 
+	var updatedOrCreatedVideoBlock []uint
+
+	ratingIds := make([]uint, 0)
+	claimantIds := make([]uint, 0)
+	rMap := make(map[uint]uint)
+	cMap := make(map[uint]uint)
+
+	for _, each := range videoData.Claims {
```
Comment (hetjagani) [inline server/action/video/update.go:287]: What is use of rMap and cMap. Can't we add directly to ratingIds & claimantIds in one loop.

File: server/util/dega.go [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+
+		isContains := contains(factCheck, action)
+
+		if viper.GetBool("dega_integration") && action == "videos" {
+
+			ctx := r.Context()
+
+			space, err := GetDegaSpace(r.Header)
+
+			log.Println("space, err", space, err)
```
Comment (hetjagani) [inline server/util/dega.go]: remove print

File: server/util/dega.go [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+
+			service := "core"
+
+			if isContains {
+				service = "fact-check"
+			}
+
+			url := fmt.Sprint(viper.GetString("dega_url"), "/", service, r.URL.Path)
+
+			req, _ := http.NewRequest(r.Method, url, nil)
```
Comment (hetjagani) [inline server/util/dega.go]: You can use requestx.Request method to simplify things

File: server/util/dega.go [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+
+			renderx.JSON(w, resp.StatusCode, res)
+			return
+		}
+
+		h.ServeHTTP(w, r)
+	})
+}
+
+func contains(s []string, e string) bool {
```
Comment (hetjagani) [inline server/util/dega.go]: can write this function in contains.go as ContainsStr or something

File: server/util/getRatingIDs.go [hunk 7]
```diff
@@ -0,0 +1,16 @@
+package util
+
+import "github.com/factly/vidcheck/model"
+
+func GetRatingAndClaimantIDs(claims []model.ClaimData) ([]uint, []uint) {
```
Comment (hetjagani) [inline server/util/getRatingIDs.go]: This function is not used anywhere in video/create and video/update you can use this.

File: server/action/video/create.go (hunk 1 above)
Comment (shreeharsha-factly) [reply server/action/video/create.go:396]: `rMap` is just used to remove repeated ratings as a `map`.  Instead of writing a nested loop stored it in map.

**Review by hetjagani — APPROVED** [review]

**Pull request merged**